platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -77,15 +77,25 @@ export default class HangingProtocolService extends PubSubService {
     },
     ModalitiesInStudy: {
       name: 'Gets the array of the modalities for the series',
-      callback: metadata =>
-        metadata.ModalitiesInStudy ??
-        (metadata.series || []).reduce((prev, curr) => {
-          const { Modality } = curr;
-          if (Modality && prev.indexOf(Modality) == -1) {
-            prev.push(Modality);
-          }
-          return prev;
-        }, []),
+      callback: metadata => {
+        if (Array.isArray(metadata.ModalitiesInStudy) && metadata.ModalitiesInStudy.length > 0) {
+          return metadata.ModalitiesInStudy;
+        }
+        if (Array.isArray(metadata.series)) {
```

Comment:
Can you extract the logic for modalities in study calculation to a utility?  Really, no one cares about it beyond it beyond the name, so it reduces the complexity of HangingProtocolService to not have it calculated inline.
Also,
metadata.series.map(ser => ser.Modality).reduce( ... ) makes it a bit clearer.